### ISSUE TITLE
`import cma` changes matplotlib behavior

### ISSUE THREAD

**Issue by MichaelClerx** [CMA-ES/pycma#17]
The following script, run from the console, produces a simple graph

```
#!/usr/bin/env python
import numpy as np
import matplotlib.pyplot as pl

#import cma

x = np.linspace(0, 10, 1000)
y = np.sin(x)

pl.figure()
pl.plot(x, y)
pl.show()
```

But when I uncomment the `cma` import, it stops producing any output!

**Comment by MichaelClerx** [comment]
Things like this: https://github.com/CMA-ES/pycma/blob/master/cma/evolution_strategy.py#L195 change Matplotlib behavior in ways that may be inconsistent with what programs _using_ pycma want to do.

**Comment by MichaelClerx** [comment]
Ok, this fixes it in a scrtip:

```
#!/usr/bin/env python
import numpy as np
import matplotlib.pyplot as pl

matplotlib_was_interactive = pl.isinteractive()
import cma
if not matplotlib_was_interactive:
    pl.ioff()

x = np.linspace(0, 10, 1000)
y = np.sin(x)

pl.figure()
pl.plot(x, y)
pl.show()
```

In general, `matplotlib.pyplot` shouldn't be imported outside of plotting methods, because users may want to configure the matplotlib backend, and this can't be done once `pyplot` has been imported (see pt 4 here: https://matplotlib.org/faq/usage_faq.html#what-is-a-backend).

I don't know what setting interactive mode on was meant to achieve, but it should probably be done in a test script rather than in one of the central modules!

**Comment by nikohansen** [comment]
The code says
```
    pyplot.ion()  # prevents that the execution stops after plotting
```
but I assume this shouldn't be necessary anymore, due to updates in `matplotlib` and/or the implementation of the plotting function in the `cma` module. Will check it out.

**Comment by MichaelClerx** [comment]
Any updates? Let me know if I can help!

**Comment by nikohansen** [comment]
The code line `pyplot.ion()` directly after the import has now been removed. However this doesn't solve the problem. I will now look into postponing the import of `pyplot` until it is actually used.

**Comment by nikohansen** [comment]
Solved with commit a121b33a3e1fc2a061b0351d2117932d08cc148a.
